pytrainer/lib/gpx.py

```diff
@@ -210,9 +217,9 @@ def getLaps(self):
                         for summary_element in lap_summary:
                             summary_dict[summary_element.get('name')] = summary_element.text
                             logging.debug("%s: %s" % (summary_element.get('name'), summary_element.text))
-                    max_speed = summary_dict["MaximumSpeed"]
-                    avg_hr = summary_dict["AverageHeartRateBpm"]
-                    max_hr = summary_dict["MaximumHeartRateBpm"]
+                    max_speed = float(summary_dict["MaximumSpeed"])
+                    avg_hr = float(summary_dict["AverageHeartRateBpm"])
+                    max_hr = float(summary_dict["MaximumHeartRateBpm"])
                 else:
```

Comment:
avg_hr and max_hr need to be int, not float (see the definition of Lap in pytrainer/core/activity.py). Also they are set in other places as well, lines 312 and 438 at least.

Also there are places where stLat and stLon are set without casting to float.